dyn_em/module_diffusion_em.F

```diff
@@ -5182,44 +5186,41 @@ SUBROUTINE nonlocal_flux (config_flags,nlflux,gamu,gamv,hpbl,kpbl,     &
   ENDDO
   ENDDO
   ENDDO
- 
-  DO j = j_start, j_end
-  DO i = i_start, i_end
-         deltaoh(i,j) = deltaoh(i,j)/hpbl(i,j)
-  ENDDO
-  ENDDO
 
   DO j = j_start, j_end
   DO i = i_start, i_end
-     delxy = sqrt(dx/msftx(i,j)*dy/msfty(i,j))
-     mlfrac        = mltop-deltaoh(i,j)
-     ezfrac        = mltop+deltaoh(i,j)
-     zfacmf(i,1,j) = min(max((zq(i,2,j)/hpbl(i,j)),zfmin),1.)
-     sfcfracn      = max(sfcfracn1,zfacmf(i,1,j))
-!
-     sflux0      = (a11+a12*sfcfracn)*sflux(i,j)
-     snlflux0    = nlfrac*sflux0
-     amf1        = snlflux0/sfcfracn
-     amf2        = -snlflux0/(mlfrac-sfcfracn)
-     bmf2        = -mlfrac*amf2
-     amf3        = snlflux0*enlfrac2(i,j)/deltaoh(i,j)
-     bmf3        = -amf3*mlfrac
-     hfxpbl(i,j) = amf3+bmf3
-     pth1        = pthnl(delxy,hpbl(i,j))
-     hfxpbl(i,j) = hfxpbl(i,j)*pth1
+     IF (pblflg(i,j)) THEN
+         deltaoh(i,j) = deltaoh(i,j)/hpbl(i,j)
+         delxy = sqrt(dx/msftx(i,j)*dy/msfty(i,j))
+         mlfrac        = mltop-deltaoh(i,j)
+         ezfrac        = mltop+deltaoh(i,j)
+         zfacmf(i,1,j) = min(max((zq(i,2,j)/hpbl(i,j)),zfmin),1.)
+         sfcfracn      = max(sfcfracn1,zfacmf(i,1,j))
+    !
+         sflux0      = (a11+a12*sfcfracn)*sflux(i,j)
+         snlflux0    = nlfrac*sflux0
+         amf1        = snlflux0/sfcfracn
+         amf2        = -snlflux0/(mlfrac-sfcfracn)
+         bmf2        = -mlfrac*amf2
+         amf3        = snlflux0*enlfrac2(i,j)/deltaoh(i,j)
+         bmf3        = -amf3*mlfrac
+         hfxpbl(i,j) = amf3+bmf3
+         pth1        = pthnl(delxy,hpbl(i,j))
+         hfxpbl(i,j) = hfxpbl(i,j)*pth1
 
-    DO k = kts, ktf
-       zfacmf(i,k,j) = max((zq(i,k+1,j)/hpbl(i,j)),zfmin)
-       IF(pblflg(i,j).and.k.LT.kpbl(i,j)) THEN
-         IF(zfacmf(i,k,j).LE.sfcfracn) THEN
-           nlflux(i,k,j) =  amf1*zfacmf(i,k,j)
-         ELSE IF (zfacmf(i,k,j).LE.mlfrac) THEN
-           nlflux(i,k,j) =  amf2*zfacmf(i,k,j)+bmf2
-         ENDIF
-         nlflux(i,k,j) = nlflux(i,k,j) + hfxpbl(i,j)*exp(-entfacmf(i,k,j)) 
-         nlflux(i,k,j) = nlflux(i,k,j)*pth1
-       ENDIF
-    ENDDO
+        DO k = kts, ktf
+           zfacmf(i,k,j) = max((zq(i,k+1,j)/hpbl(i,j)),zfmin)
+           IF(k.LT.kpbl(i,j)) THEN
+             IF(zfacmf(i,k,j).LE.sfcfracn) THEN
+               nlflux(i,k,j) =  amf1*zfacmf(i,k,j)
+             ELSE IF (zfacmf(i,k,j).LE.mlfrac) THEN
+               nlflux(i,k,j) =  amf2*zfacmf(i,k,j)+bmf2
+             ENDIF
+             nlflux(i,k,j) = nlflux(i,k,j) + hfxpbl(i,j)*exp(-entfacmf(i,k,j))
+             nlflux(i,k,j) = nlflux(i,k,j)*pth1
+           ENDIF
+        ENDDO
+    ENDIF
```

Comment:
we need hpbl and deltaoh to be non-zero for this section to work without errors. This means pblflg needs to be .true. If pblflg is .false. this section doesn't affect the output anyway.
Note that I merge two neighbouring i,j loops here.

Reply:
Thanks for your fix. I agree with your fix.